tests/test_ext_autodoc_preserve_defaults.py

```diff
@@ -0,0 +1,45 @@
+"""
+    test_ext_autodoc_preserve_defaults
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Test the autodoc extension.
+
+    :copyright: Copyright 2007-2021 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import pytest
+
+from .test_ext_autodoc import do_autodoc
+
+
+@pytest.mark.sphinx('html', testroot='ext-autodoc',
+                    confoverrides={'extensions': ['sphinx.ext.autodoc.preserve_defaults']})
```

Comment:
At present, this extension is enabled on loading. To integrate this into the autodoc core, configuration must be needed. How do we do?

Reply:
Finally, I chose the way to enable this by configuration instead of loading the extension.